Agoric/Agoric
# test(a3p): test mintHolder functionalities after contract null upgrade

closes: #10410 

## Description


This pull request intends to ensure that the mintHolder contract continues to function as expected after a null upgrade. 

Key Changes:
- A new `core-eval` was built, which acordingly to the agoric chain variant, will select a list of vbankAsset and execute a null upgrade to the respective mintHolder contract of each suitable asset.
- A new acceptance test, `mintHolder.test.js`, has been added to the a3p-integration to verify the core-eval behaviour.

MintHolder Test Plan:
   - Provision a receiver wallet with the respective assets
   - Confirm the initial balance matches expectations
   - Perform a null upgrade for the mintHolder contracts
   - Execute a core-eval to:
         - Mint a payment for each selected asset
   	- Deposit the payment into the receiver's depositFacet
   - Verify that the balance has increased by the expected amount
   - Execute a PSM swap for selected assets with IST


### Security Considerations


The execution of `mintHolder.test.js` had to be done prior to the `./genesis-test.sh` acceptance test. Otherwise it would trigger the error `bundle ${id} not loaded` during the `evalBundle` call.
See this issue for more detail: #10620 

### Scaling Considerations


This PR does not introduce any change to the mintHolder source code, or how it is being consumed. For this reason that are no scaling considerations to be made.

### Documentation Considerations


### Testing Considerations


The implemented test design is expecting to have a core-eval for each mintHolder being updated.
If desired, this could be refactored to have a single core-eval executing a null upgrade to a predefined list of assets.
The second approach has the advantage of reducing the number of files being created, although it may hinder the testing flexibility.

Which approach is more desirable?

### Upgrade Considerations
